yellowpages.lic

```diff
@@ -0,0 +1,70 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#yellowpages
+=end
+
+arg_definitions = [
+    [
+	{ name: 'debug', regex: /debug/i, optional: true, description: 'Very verbose, may cause lich to hang for several seconds if using large contact files' },
+    { name: 'search', regex: /[a-zA-Z]/, optional: true, description: 'Name of player to look up in contacts.' }
+	]
+]
+args = parse_args(arg_definitions)
+$debug_mode_cc = args.debug
+
+#DEBUG INFO
+echo args.debug if $debug_mode_cc
+echo args.search if $debug_mode_cc
+
+#Load base contacts YAML
+Contacts = YAML.load_file("./scripts/data/base-contacts.yaml")
+Contacts = Contacts['contacts']
+
+#DEBUG INFO
+echo "---Base Contacts---" if $debug_mode_cc
+echo Contacts.inspect if $debug_mode_cc
+echo Contacts.class if $debug_mode_cc
+
+#Check for <CharName>-contacts.yaml file and merges it if exists
+if File.exist?("./scripts/profiles/#{checkname}-contacts.yaml")
+	##DEBUG
+	echo "Personal Contact file found! Now importing!" if $debug_mode_cc
+	
+	Personal_Contacts = YAML.load_file("./scripts/profiles/#{checkname}-contacts.yaml")
+	Personal_Contacts = Personal_Contacts['contacts']
+
+	#DEBUG INFO
+	echo "---Personal Contacts---" if $debug_mode_cc
+	echo Personal_Contacts.inspect if $debug_mode_cc
+	echo Personal_Contacts.class if $debug_mode_cc
+
+	Contacts.merge!(Personal_Contacts)
+end
+
+#DEBUG INFO
+echo "---Merged Contacts---" if $debug_mode_cc
+echo Contacts.inspect if $debug_mode_cc
+echo Contacts.class if $debug_mode_cc
+echo "-----" if $debug_mode_cc
+
+if args.search.nil?
+	echo "What user do you wish to search for?"

```

Comment:
I tried making search not optional first, but when I added the debug argument things wouldn't work with ;script debug name. It would only work with ;script debug OR ;script name. This was the quickest way to get it working again.